code/modules/xenomorph/resin_gargoyle.dm

```diff
@@ -0,0 +1,61 @@
+/obj/structure/xeno/resin_gargoyle
+	name = "resin gargoyle"
+	desc = "A resin monument to your tresspass. Alerts the xenomorph hive when an enemy approaches."
+	icon = 'icons/Xeno/2x2building.dmi'
+	icon_state = "gargoyle"
+	max_integrity = 100
+	///Bool if we're currently alerting
+	var/is_alerting = FALSE
+	//cd tracking for the alert
+	COOLDOWN_DECLARE(proxy_alert_cooldown)
+
+/obj/structure/xeno/resin_gargoyle/Initialize(mapload, _hivenumber, mob/living/carbon/xenomorph/creator)
+	. = ..()
+	for(var/turfs in RANGE_TURFS(XENO_GARGOYLE_DETECTION_RANGE, src))
+		RegisterSignal(turfs, COMSIG_ATOM_ENTERED, PROC_REF(gargoyle_alarm))
```

Comment:
Should unregister these signals on destroy

Reply:
https://github.com/tgstation/TerraGov-Marine-Corps/blob/e0a7b011bc5def4075ef345019d14b1e1a71b4e9/code/datums/datum.dm#L129

Shouldn't be necesarry

Reply:
yeah signals are all unregistered on destroy already.